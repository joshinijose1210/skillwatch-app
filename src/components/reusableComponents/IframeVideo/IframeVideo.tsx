const IframeVideo = ({ videoId }: { videoId: string }) => {
    return (
        <iframe
            title="YouTube video"
            loading="lazy"
            allowFullScreen
            src={`https://www.youtube.com/embed/${videoId}/?controls=1&autoplay=1&rel=0`}
            style={{
                border: 'none',
                borderRadius: '10px',
                width: '100%',
                height: '100%',
                minHeight: 0,
                minWidth: 0
            }}
            allow="accelerometer; autoplay; clipboard-write; encrypted-media;"
            referrerPolicy="strict-origin-when-cross-origin"
        />
    );
};

export default IframeVideo;
