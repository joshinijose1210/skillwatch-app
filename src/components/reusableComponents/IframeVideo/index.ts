import IframeVideo from './IframeVideo';

export default IframeVideo;
