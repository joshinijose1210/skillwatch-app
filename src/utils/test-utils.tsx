import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
