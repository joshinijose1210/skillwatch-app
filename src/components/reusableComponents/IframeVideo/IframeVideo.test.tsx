import { screen } from '@testing-library/react';
import { RTKRender } from '@utils/test-utils';
import IframeVideo from './IframeVideo';

const renderer = (videoId = 'dQw4w9WgXcQ') => RTKRender(<IframeVideo videoId={videoId} />);

describe('IframeVideo', () => {
    it('should render iframe with correct YouTube embed src', () => {
        const testVideoId = 'test123';
        RTKRender(<IframeVideo videoId={testVideoId} />);

        const iframe = screen.getByTitle('YouTube video');
        expect(iframe).toBeInTheDocument();
        expect(iframe).toHaveAttribute('src', `https://www.youtube.com/embed/${testVideoId}/?controls=1&autoplay=1&rel=0`);
    });

    it('should match snapshot', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });
});
