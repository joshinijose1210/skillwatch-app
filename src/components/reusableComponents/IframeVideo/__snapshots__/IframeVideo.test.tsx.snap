// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IframeVideo should match snapshot 1`] = `
<div>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media;"
    allowfullscreen=""
    loading="lazy"
    referrerpolicy="strict-origin-when-cross-origin"
    src="https://www.youtube.com/embed/dQw4w9WgXcQ/?controls=1&autoplay=1&rel=0"
    style="border-radius: 10px; width: 100%; height: 100%; min-height: 0; min-width: 0;"
    title="YouTube video"
  />
</div>
`;
